File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1, r = *max_element(piles.begin(), piles.end()), min_rate = r;
        while (l <= r)
        {
            int k = l + (r - l) / 2;
            long long totalTime = 0;
            for (int p : piles) 
                totalTime += ceil(static_cast<double>(p) / k);
            if (totalTime <= h) 
            {
                min_rate = k;
                r = k - 1;
            } 
            else
                l = k + 1;

        }
        return min_rate;
        
    }
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1
        r = max(piles)
        min_k = r
        while l <= r:
            mid = int(l + ( r - l ) / 2)
            total_time = 0
            for p in piles:
                total_time += math.ceil(float(p) / mid)
            if total_time <= h:
                min_k = mid
                r = mid - 1
            else:
                l = mid + 1
        return min_k

